iidm/iidm-api/src/main/java/com/powsybl/iidm/network/ShuntCompensatorAdder.java

```diff
@@ -26,9 +26,9 @@
  */
 public interface ShuntCompensatorAdder extends InjectionAdder<ShuntCompensatorAdder> {
 
-    ShuntCompensatorAdder setbPerSection(double bPerSection);
+    ShuntCompensatorLinearModelAdder newLinearModel();
 
-    ShuntCompensatorAdder setMaximumSectionCount(int maximumSectionCount);
+    ShuntCompensatorNonLinearModelAdder newNonLinearModel();
```

Comment:
To be discussed: should we design the API like we did for the extension:
```
newModel(ShuntCompensatorLinearModelAdder.class);
```

Reply:
Do you think that choosing the same design here could improve the user experience?

Reply:
It is mostly to keep an easy way to add a custom model... but doesn't it make the API more complex? (as in, it may make it more difficult to understand what "model" refers to)? Maybe we should keep `newLinearModel`, `newNonLinearModel` and add `newModel(...)`?

Reply:
After some thinking, I do not think it is a good idea to implement a `newModel(...)` method, I am not sure we want to extend the possibilities of shunt models.